File: web/src/api/sysDictionary.ts
import service from '@/utils/request'

export interface SysDictionary {
    ID?: number
    name?: string
    type?: string
    status?: boolean
    desc?: string
    sysDictionaryDetails?: any[]
}

export interface SysDictionarySearch {
    page: number
    pageSize: number
    name?: string
    type?: string
    status?: boolean
    desc?: string
}

// @Tags SysDictionary
// @Summary 创建SysDictionary
export const createSysDictionary = (data: SysDictionary) => {
    return service({
        url: '/api/v1/dictionary/add',
        method: 'post',
        data
    })
}

// @Tags SysDictionary
// @Summary 删除SysDictionary
export const deleteSysDictionary = (data: { ID: number }) => {
    return service({
        url: '/api/v1/dictionary/delete',
        method: 'post',
        data
    })
}

// @Tags SysDictionary
// @Summary 更新SysDictionary
export const updateSysDictionary = (data: SysDictionary) => {
    return service({
        url: '/api/v1/dictionary/update',
        method: 'post',
        data
    })
}

// @Tags SysDictionary
// @Summary 用id查询SysDictionary
export const findSysDictionary = (params: { ID: number | string }) => {
    return service({
        url: '/api/v1/dictionary/get',
        method: 'get',
        params
    })
}

// @Tags SysDictionary
// @Summary 分页获取SysDictionary列表
export const getSysDictionaryList = (params: SysDictionarySearch) => {
    return service({
        url: '/api/v1/dictionary/list',
        method: 'get',
        params
    })
}

// @Tags SysDictionary
// @Summary 导出字典JSON（包含字典详情）
export const exportSysDictionary = (params?: { ID: number | string }) => {
    return service({
        url: '/api/v1/dictionary/export',
        method: 'get',
        params
    })
}

// @Tags SysDictionary
// @Summary 导入字典JSON（包含字典详情）
export const importSysDictionary = (data: any) => {
    return service({
        url: '/api/v1/dictionary/import',
        method: 'post',
        data
    })
}

File: server/router/system/sys_dictionary.go
package system

import (
	"gin-vue-admin/middleware"

	"github.com/gin-gonic/gin"
)

type DictionaryRouter struct{}

func (s *DictionaryRouter) InitSysDictionaryRouter(Router *gin.RouterGroup) {
	sysDictionaryRouter := Router.Group("dictionary").Use(middleware.OperationRecord())
	{
		sysDictionaryRouter.POST("add", dictionaryApi.CreateSysDictionary)    // 新建SysDictionary
		sysDictionaryRouter.POST("delete", dictionaryApi.DeleteSysDictionary) // 删除SysDictionary
		sysDictionaryRouter.POST("update", dictionaryApi.UpdateSysDictionary) // 更新SysDictionary
		sysDictionaryRouter.POST("import", dictionaryApi.ImportSysDictionary) // 导入SysDictionary
		sysDictionaryRouter.GET("export", dictionaryApi.ExportSysDictionary)  // 导出SysDictionary
		sysDictionaryRouter.GET("get", dictionaryApi.FindSysDictionary)       // 根据ID获取SysDictionary
		sysDictionaryRouter.GET("list", dictionaryApi.GetSysDictionaryList)   // 获取SysDictionary列表
	}
}
